File: backend/app/models/artwork.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, Text
from sqlalchemy.orm import relationship
from datetime import datetime
from ..core.database import Base


class Artwork(Base):
    """Model for storing user drawings/artwork"""
    __tablename__ = "artworks"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False, index=True)
    title = Column(String(255), nullable=True)
    image_data = Column(Text, nullable=False)  # Base64 encoded PNG image
    thumbnail_data = Column(Text, nullable=True)  # Base64 encoded thumbnail
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    updated_at = Column(DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    user = relationship("User", back_populates="artworks")

    def __repr__(self):
        return f"<Artwork(id={self.id}, user_id={self.user_id}, title={self.title})>"

File: mental-health-chatbot/frontend/lib/core/models/artwork.dart
class ArtworkModel {
  final int? id;
  final int? userId;
  final String? title;
  final String imageData;
  final String? thumbnailData;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  ArtworkModel({
    this.id,
    this.userId,
    this.title,
    required this.imageData,
    this.thumbnailData,
    this.createdAt,
    this.updatedAt,
  });

  factory ArtworkModel.fromJson(Map<String, dynamic> json) {
    return ArtworkModel(
      id: json['id'] as int?,
      userId: json['user_id'] as int?,
      title: json['title'] as String?,
      imageData: json['image_data'] as String,
      thumbnailData: json['thumbnail_data'] as String?,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      updatedAt: json['updated_at'] != null
          ? DateTime.parse(json['updated_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'id': id,
      if (userId != null) 'user_id': userId,
      if (title != null) 'title': title,
      'image_data': imageData,
      if (thumbnailData != null) 'thumbnail_data': thumbnailData,
    };
  }

  ArtworkModel copyWith({
    int? id,
    int? userId,
    String? title,
    String? imageData,
    String? thumbnailData,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return ArtworkModel(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      title: title ?? this.title,
      imageData: imageData ?? this.imageData,
      thumbnailData: thumbnailData ?? this.thumbnailData,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

class ArtworkThumbnail {
  final int id;
  final String? title;
  final String? thumbnailData;
  final DateTime createdAt;

  ArtworkThumbnail({
    required this.id,
    this.title,
    this.thumbnailData,
    required this.createdAt,
  });

  factory ArtworkThumbnail.fromJson(Map<String, dynamic> json) {
    return ArtworkThumbnail(
      id: json['id'] as int,
      title: json['title'] as String?,
      thumbnailData: json['thumbnail_data'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
    );
  }
}
